algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -73,16 +95,169 @@ class RandomAccessIterator< ::Kokkos::View<DataType, Args...> > {
+  iterator_type operator+(difference_type n) const {
+    auto it = *this;
+    it += n;
+    return it;
+  }
+
+  KOKKOS_FUNCTION
+  iterator_type operator-(difference_type n) const {
+    auto it = *this;
+    it -= n;
+    return it;
+  }
+
+  KOKKOS_FUNCTION
+  difference_type operator-(iterator_type it) const {
+    if constexpr (is_always_contiguous)
+      return m_data - it.m_data;
+    else
+      return (m_data - it.m_data) / m_stride;
+  }
+
+  KOKKOS_FUNCTION
+  bool operator==(iterator_type other) const { return m_data == other.m_data; }
+
+  KOKKOS_FUNCTION
+  bool operator!=(iterator_type other) const { return m_data != other.m_data; }
+
+  KOKKOS_FUNCTION
+  bool operator<(iterator_type other) const { return m_data < other.m_data; }
+
+  KOKKOS_FUNCTION
+  bool operator<=(iterator_type other) const { return m_data <= other.m_data; }
+
+  KOKKOS_FUNCTION
+  bool operator>(iterator_type other) const { return m_data > other.m_data; }
+
+  KOKKOS_FUNCTION
+  bool operator>=(iterator_type other) const { return m_data >= other.m_data; }
+
+  KOKKOS_FUNCTION
+  reference operator*() const { return *m_data; }
+
+  KOKKOS_FUNCTION
+  pointer data() const { return m_data; }
+
+  KOKKOS_FUNCTION
+  int stride() const { return m_stride; }
+
+ private:
+  pointer m_data;
+  int m_stride;
+  static constexpr bool is_always_contiguous =
+      (std::is_same_v<typename view_type::traits::array_layout,
+                      Kokkos::LayoutLeft> ||
+       std::is_same_v<typename view_type::traits::array_layout,
+                      Kokkos::LayoutRight>);
+
+  // Needed for the converting constructor accepting another iterator
+  template <class, class>
+  friend class RandomAccessIterator;
+};
+
+template <class DataType, class... Args>
+class RandomAccessIterator<::Kokkos::View<DataType, Args...>,
+                           std::enable_if_t<!is_always_strided<
+                               ::Kokkos::View<DataType, Args...>>::value>> {
+ public:
+  using view_type     = ::Kokkos::View<DataType, Args...>;
+  using iterator_type = RandomAccessIterator<view_type>;
+
+  using iterator_category = std::random_access_iterator_tag;
+  using value_type        = typename view_type::value_type;
+  using difference_type   = ptrdiff_t;
+  using pointer           = typename view_type::pointer_type;
+  using reference         = typename view_type::reference_type;
+
+  static_assert(view_type::rank == 1 &&
+                    (std::is_same<typename view_type::traits::array_layout,
+                                  Kokkos::LayoutLeft>::value ||
+                     std::is_same<typename view_type::traits::array_layout,
+                                  Kokkos::LayoutRight>::value ||
+                     std::is_same<typename view_type::traits::array_layout,
+                                  Kokkos::LayoutStride>::value),
+                "RandomAccessIterator only supports 1D Views with LayoutLeft, "
+                "LayoutRight, LayoutStride.");
```

Comment:
```suggestion
  static_assert(view_type::rank == 1,
                "RandomAccessIterator only supports 1D Views");
```
In the current form, the `static_assert` should trigger for non-strided layouts since the ones checked here are all strided.

Reply:
You are right.